movie-service/src/main/java/com/garret/movies/service/impl/MovieServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+package com.garret.movies.service.impl;
+
+import com.garret.movies.dao.entity.Movie;
+import com.garret.movies.dao.repository.MovieRepository;
+import com.garret.movies.service.api.MovieService;
+import com.garret.movies.service.exception.IncorrectDateException;
+import com.google.inject.internal.util.ImmutableList;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+@AllArgsConstructor
+public class MovieServiceImpl implements MovieService {
+
+    private MovieRepository movieRepository;
+
+    @Override
+    @Transactional
+    public Movie save(@NonNull Movie movie) {
+        Optional<Movie> optionalMovie = movieRepository.findByImdbId(movie.getImdbId());
+        if (optionalMovie.isPresent()) {
```

Comment:
@Garret-vin In most of cases Optional::get can be avoided, try to use build in mehods of optional to transform data.
In our case whole this method code can be replaced with:

`
return movieRepository.findByImdbId(movie.getImdbId()).orElseGet(() -> movieRepository.save(movie));
`